Content.Server/Access/Systems/IdCardConsoleSystem.cs

```diff
@@ -151,25 +151,32 @@ private void TryWriteToTargetId(EntityUid uid,
         if (oldTags.SequenceEqual(newAccessList))
             return;
 
-        // I hate that C# doesn't have an option for this and don't desire to write this out the hard way.
-        // var difference = newAccessList.Difference(oldTags);
-        var difference = newAccessList.Union(oldTags).Except(newAccessList.Intersect(oldTags)).ToHashSet();
+        var ableToModify = component.AccessLevels.ToHashSet();
+
+        var addedTags = newAccessList.Except(oldTags).Select(tag => tag).ToHashSet();
+        var removedTags = oldTags.Except(newAccessList).Select(tag => tag).ToHashSet();
+
+        //Probably there is a cleaner way to do this, visible difference is the difference between the new access and the old access the user of the computer could see
+        var difference = addedTags.Union(removedTags).ToHashSet();
+        var visibleDifference = difference.Intersect(ableToModify).ToHashSet();
+        var hiddenAccess = difference.Except(visibleDifference).ToList();
+
         // NULL SAFETY: PrivilegedIdIsAuthorized checked this earlier.
         var privilegedPerms = _accessReader.FindAccessTags(privilegedId!.Value).ToHashSet();
-        if (!difference.IsSubsetOf(privilegedPerms))
+        if (!visibleDifference.IsSubsetOf(privilegedPerms))
         {
             _sawmill.Warning($"User {ToPrettyString(uid)} tried to modify permissions they could not give/take!");
             return;
         }
 
-        var addedTags = newAccessList.Except(oldTags).Select(tag => "+" + tag).ToList();
-        var removedTags = oldTags.Except(newAccessList).Select(tag => "-" + tag).ToList();
-        _access.TrySetTags(targetId, newAccessList);
+        _access.TrySetTags(targetId, newAccessList.Union(hiddenAccess));
```

Comment:
The way this code works is extremely non-obvious, and I had to check your code out in a debugger to even understand it. You are implicitly relying on `hiddenAccess` only ever containing negative differences due to removal of these tags by the UI, which works, but is extremely unclear.

Please either restructure this code to make the behavior more obvious, or make it heavily commented.